src/main/java/org/stellar/sdk/SorobanServer.java

```diff
@@ -0,0 +1,344 @@
+            .accountID(KeyPair.fromAccountId(accountId).getXdrAccountId())
+            .build();
+    LedgerKey ledgerKey =
+        new LedgerKey.Builder()
+            .account(ledgerKeyAccount)
+            .discriminant(LedgerEntryType.ACCOUNT)
+            .build();
+    GetLedgerEntriesResponse getLedgerEntriesResponse =
+        this.getLedgerEntries(Collections.singleton(ledgerKey));
+    List<GetLedgerEntriesResponse.LedgerEntryResult> entries =
+        getLedgerEntriesResponse.getEntries();
+    if (entries == null || entries.isEmpty()) {
+      throw new LedgerEntryNotFoundException(ledgerKeyToXdrBase64(ledgerKey));
+    }
+    LedgerEntry.LedgerEntryData ledgerEntryData =
+        ledgerEntryDataFromXdrBase64(entries.get(0).getXdr());
+    long sequence = ledgerEntryData.getAccount().getSeqNum().getSequenceNumber().getInt64();
+    return new Account(accountId, sequence);
+  }
+
+  public GetHealthResponse getHealth() throws IOException {
+    return this.<Void, GetHealthResponse>sendRequest(
+        "getHealth", null, new TypeToken<SorobanRpcResponse<GetHealthResponse>>() {});
+  }
+
+  public GetLedgerEntriesResponse.LedgerEntryResult getContractData(
+      String contractId, SCVal key, Durability durability)
+      throws IOException, LedgerEntryNotFoundException {
+
+    ContractDataDurability contractDataDurability;
+    switch (durability) {
+      case TEMPORARY:
+        contractDataDurability = ContractDataDurability.TEMPORARY;
+        break;
+      case PERSISTENT:
+        contractDataDurability = ContractDataDurability.PERSISTENT;
+        break;
+      default:
+        throw new IllegalArgumentException("Invalid durability: " + durability);
+    }
+
+    Address address = new Address(contractId);
+    LedgerKey.LedgerKeyContractData ledgerKeyContractData =
+        new LedgerKey.LedgerKeyContractData.Builder()
+            .contract(address.toSCAddress())
+            .key(key)
+            .durability(contractDataDurability)
+            .bodyType(ContractEntryBodyType.DATA_ENTRY)
+            .build();
+    LedgerKey ledgerKey =
+        new LedgerKey.Builder()
+            .discriminant(LedgerEntryType.CONTRACT_DATA)
+            .contractData(ledgerKeyContractData)
+            .build();
+    GetLedgerEntriesResponse getLedgerEntriesResponse =
+        this.getLedgerEntries(Collections.singleton(ledgerKey));
+    List<GetLedgerEntriesResponse.LedgerEntryResult> entries =
+        getLedgerEntriesResponse.getEntries();
+    if (entries == null || entries.isEmpty()) {
+      throw new LedgerEntryNotFoundException(ledgerKeyToXdrBase64(ledgerKey));
+    }
+
+    return entries.get(0);
+  }
+
+  public GetLedgerEntriesResponse getLedgerEntries(Collection<LedgerKey> keys) throws IOException {
+    List<String> xdrKeys =
+        keys.stream().map(SorobanServer::ledgerKeyToXdrBase64).collect(Collectors.toList());
+    GetLedgerEntriesRequest params = new GetLedgerEntriesRequest(xdrKeys);
```

Comment:
is an empty `keys` list a valid  request and/or should this check for empty list on `xdrKeys`, to avoid sending to rpc in any case?

Reply:
Fixed.